dhis-2/dhis-services/dhis-service-dxf2/src/main/java/org/hisp/dhis/dxf2/metadata/objectbundle/validation/ValidationUtils.java

```diff
@@ -36,39 +38,42 @@
 import org.hisp.dhis.dxf2.metadata.objectbundle.ObjectBundle;
 import org.hisp.dhis.feedback.ErrorReport;
 import org.hisp.dhis.feedback.ObjectReport;
-import org.hisp.dhis.feedback.TypeReport;
 
 /**
  * @author Luciano Fiandesio
  */
 public class ValidationUtils
 {
+    private ValidationUtils()
+    {
+        throw new UnsupportedOperationException( "util" );
+    }
 
-    public static List<IdentifiableObject> joinObjects( List<IdentifiableObject> persistedObjects,
-        List<IdentifiableObject> nonPersistedObjects )
+    public static <T> List<T> joinObjects( List<T> persistedObjects,
+        List<T> nonPersistedObjects )
     {
+        if ( persistedObjects.isEmpty() )
+        {
+            return nonPersistedObjects;
+        }
+        if ( nonPersistedObjects.isEmpty() )
+        {
+            return persistedObjects;
+        }
         return Stream.concat( persistedObjects.stream(), nonPersistedObjects.stream() ).collect( Collectors.toList() );
     }
 
-    public static void addObjectReports( List<ErrorReport> reports, TypeReport typeReport, IdentifiableObject object,
+    public static ObjectReport createObjectReport( List<ErrorReport> reports, IdentifiableObject object,
         ObjectBundle bundle )
     {
         ObjectReport objectReport = new ObjectReport( object, bundle );
         objectReport.setDisplayName( IdentifiableObjectUtils.getDisplayName( object ) );
         objectReport.addErrorReports( reports );
-
-        typeReport.addObjectReport( objectReport );
-        typeReport.getStats().incIgnored();
```

Comment:
FYI: this now happens in the `ObjectValidationCheck` when a created `ObjectReport` is consumed.